packages/frontend/src/hooks/useBigNumberConfig.ts

```diff
@@ -0,0 +1,53 @@
+import {
+    ApiQueryResults,
+    BigNumber,
+    Explore,
+    findFieldByIdInExplore,
+    friendlyName,
+    getFieldLabel,
+} from 'common';
+import { useCallback, useEffect, useState } from 'react';
+
+const useBigNumberConfig = (
+    bigNumberConfigData: BigNumber | undefined,
+    resultsData: ApiQueryResults | undefined,
+    explore: Explore | undefined,
+) => {
+    const metric = resultsData?.metricQuery.metrics[0];
+    const bigNumberValue =
+        metric && resultsData?.rows?.[0][metric].value.formatted;
+
+    const fieldId = resultsData?.metricQuery.metrics[0];
+
+    const [bigNumber, setBigNumber] = useState<string | number>('');
```

Comment:
Do you need this state? 

Could you just return `bigNumberValue` as `bigNumber`?